Common++/header/IpUtils.h

```diff
@@ -62,22 +63,48 @@ namespace pcpp
 		 * Extract IPv4 address from sockaddr
 		 * @param[in] sa - input sockaddr
 		 * @return Address in in_addr format
+		 * @throws std::invalid_argument Sockaddr family is not AF_INET or sockaddr is nullptr.
 		 */
-		in_addr* sockaddr2in_addr(struct sockaddr *sa);
+		in_addr* sockaddr2in_addr(sockaddr* sa);
+
+		/**
+		 * Attempt to extract IPv4 address from sockaddr
+		 * @param[in] sa - input sockaddr
+		 * @return Pointer to address in in_addr format or nullptr if extraction fails.
+		 */
+		in_addr* try_sockaddr2in_addr(sockaddr* sa);
 
 		/**
 		 * Extract IPv6 address from sockaddr
 		 * @param[in] sa - input sockaddr
 		 * @return Address in in6_addr format
+		 * @throws std::invalid_argument Sockaddr family is not AF_INET6 or sockaddr is nullptr.
+		 */
+		in6_addr* sockaddr2in6_addr(sockaddr* sa);
+
+		/**
+		 * Attempt to extract IPv6 address from sockaddr
+		 * @param[in] sa - input sockaddr
+		 * @return Pointer to address in in6_addr format or nullptr if extraction fails.
 		 */
-		in6_addr* sockaddr2in6_addr(struct sockaddr *sa);
+		in6_addr* try_sockaddr2in6_addr(sockaddr* sa);
 
 		/**
 		 * Converts a sockaddr format address to its string representation
 		 * @param[in] sa Address in sockaddr format
-		 * @param[out]  resultString String representation of the address
+		 * @param[out] resultString String representation of the address
+		 * @param[in] resultBufLen Length of the result buffer.
+		 * @throws std::invalid_argument Sockaddr family is not AF_INET or AF_INET6, sockaddr is nullptr or the result str buffer is insufficient.
+		 */
+		void sockaddr2string(sockaddr const* sa, char* resultString, size_t resultBufLen);
+
+		/**
+		 * Converts a spckaddr format address to its string representation
+		 * @param[in] sa Address in sockaddr format
+		 * @return String representation of the address
+		 * @throws std::invalid_argument Sockaddr family is not AF_INET or AF_INET6, or sockaddr is nullptr.
 		 */
-		void sockaddr2string(struct sockaddr *sa, char* resultString);
+		std::string sockaddr2string(sockaddr const* sa);
```

Comment:
do we use this overload of `sockaddr2string` anywhere? 🤔

Reply:
Atm, not really, but I thought it was a good overload to have, in case of just needing a `string`.

Reply:
These are internal APIs that are not used a lot. I'd suggest reducing it to only what we need

Reply:
Removed 202fa55

Reply:
Fixed 202fa55 and 518b837.

Accidentally removed the wrong function from the header in the first commit, whops.